mesonbuild/build.py

```diff
@@ -1240,6 +1237,12 @@ def get_clink_dynamic_linker_and_stdlibs(self):
         # Populate list of all compilers, not just those being used to compile
         # sources in this target
         all_compilers = self.environment.coredata.compilers[self.for_machine]
+
+        # If the user set the link_language, just return that.
+        if self.link_language:
+            comp = all_compilers[self.link_language]
+            return comp, comp.language_stdlib_only_link_flags()
+
```

Comment:
Why not something like

```python
        # did user override clink_langs for this target?
        link_langs = [self.link_language] if self.link_language else clink_langs
```
and use `link_langs` below instead of `clink_langs` ? Nothing wrong with code above but code below returns a more friendly error in case the compiler was not found.

Reply:
No, that change I proposed is not enough. I would need to skip the part of code calling `language_stdlib_only_link_flags`.